app/assets/javascripts/models/beta/history.js

```diff
@@ -0,0 +1,28 @@
+import { sortBy } from 'underscore';
+
+export const getDate = (oldValue, newValue) =>
+  [
+    oldValue ? I18n.l("date.formats.long", oldValue) : null,
+    oldValue ? I18n.l("date.formats.long", newValue) : null
+  ];
+
+const isDate = key => key.endsWith("_at");
+
+export const deserializeChanges = changes =>
+  Object.keys(changes).map(key => {
+    const change = changes[key];
+    const [oldValue, newValue] = isDate(key) ? getDate(change[0], change[1]) : change;
+    return { oldValue, newValue, key };
+  });
+
+export const sortByActivityCreation = history =>
+  sortBy(history, ({ activity }) => activity.created_at);
+
+export const deserializeHistory = (history, users) =>
+  sortByActivityCreation(history).map(item => ({
+    activity: {
+      ...item.activity,
+      changes: deserializeChanges(item.activity.subject_changes),
+      user: users.find(user => user.id === item.activity.user_id).name
```

Comment:
What do you think  about creating a model `User` and have a method `User.findById(users, id)`?

Reply:
Sure!